crates/gradbench/src/main.rs

```diff
@@ -675,6 +701,18 @@ fn cli() -> Result<(), ExitCode> {
                 }
             }
         }
+        Commands::Log { command } => match command {
+            LogCommands::Trim { input, output } => {
+                let input_file = fs::File::open(input).map_err(|err| err_fail(anyhow!(err)))?;
+                let mut output_file: Box<dyn std::io::Write> = match output {
+                    Some(path) => {
+                        Box::new(fs::File::create(&path).map_err(|err| err_fail(anyhow!(err)))?)
+                    }
+                    None => Box::new(io::stdout()),
+                };
+                log::trim(&mut io::BufReader::new(input_file), &mut output_file).map_err(err_fail)
+            }
+        },
```

Comment:
The repeated use of `anyhow!` and `err_fail` here adds a lot of noise. The reason you found yourself having to do this is because this function returns `Result<(), ExitCode>` instead of `anyhow::Result<()>`, so you aren't really able to make effective use of the `?` operator. Could you please instead move this into a separate function that returns `anyhow::Result<()>` and just call that, then have one single usage of `.map_err(err_fail)` here?